frontend/shared/src/hooks/useGoToPage.ts

```diff
@@ -21,10 +20,12 @@ const useGoToPage = (): GoToPageFunction => {
         const op = operation ?? 'push';
         void router[op](path, asPath ?? path, routerOptions);
       } else {
-        void router.push(path);
+        // use shallow route when we are already on the same path (ignoring query params)
+        const shallow = pagePath === router.pathname;
+        void router.push(path, undefined, { shallow });
```

Comment:
If there are no options given, router is not yet routing and pagePath has query parameters but they have changed then what happens? Doesn't it first not go into the `if (isAlreadyRouting || pagePath === router.asPath)` because the router is not yet routing and the query parameters have changed, then it doesn't go into the `if (options)` clause because there are no options but into its `else` clause? And then in that part it will use shallow routing because the base pathname is the same? So doesn't this use shallow routing for an URL whose query parameters have changed in this case? Or did I miss or misinterpret something?

Reply:
The `options` were used e.g. in ApplicationPage:

```typescript
  if (applicationQuery.isSuccess) {
    if (applicationQuery.data.status !== 'draft' && applicationId) {
      void goToPage(`/thankyou?id=${applicationId}`, { operation: 'replace' });
    }
```

After discussion using the shallow routing for the otherwise same URL but with only the query parameters changed is ok, I wasn't sure whether the page contents would get updated correctly in that situation with shallow routing but they should so that's ok.

Reply:
Fixed now. Options were reduced to only allow the operation to be changed which can be  `push` or `replace`. `push` is default when operation type is not given.